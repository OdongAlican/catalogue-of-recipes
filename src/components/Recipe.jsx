import React from 'react'


function Recipe({ recipe }){
    return (
        <div>
            <div className="card col-md-3 mx-2 mb-2" >
                <ul className="list-group list-group-flush">
                    <li className="list-group-item">{ recipe.name }</li>
                    <li className="list-group-item"> { recipe.username }</li>
                    <li className="list-group-item"> { recipe.email } </li>
                </ul>
            </div>
        </div>
    )
}

export default Recipe