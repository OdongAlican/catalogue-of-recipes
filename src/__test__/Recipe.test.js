import React from 'react'
import Recipe from '../components/Recipe'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import { BrowserRouter as Router } from 'react-router-dom';

const recipe = { 
                image:"sample", 
                title:"title",
                site: "site"
               }

afterEach(cleanup)

it("renders recipe correctly", ()=>{
    const { getAllByTestId } = render(<Recipe recipe = {recipe}/>, { wrapper: MemoryRouter })
    const result = getAllByTestId('recipe')
    expect(result[0]).toHaveTextContent("title company: Visit Us:siteView Details")
})

it("another rendering of recipe correctly", ()=>{
    const { getAllByTestId } = render(<Recipe recipe = {recipe}/>, { wrapper: MemoryRouter })
    const result = getAllByTestId('recipe')
    expect(result[0]).not.toHaveTextContent("titler company: Visit Us:siteView Details")
})

it('matches recipe snapshot', () =>{
    const appointment = renderer.create(<Router><Recipe recipe = {recipe} /></Router>);
    let tree = appointment.toJSON();
    expect(tree).toMatchSnapshot();
  })