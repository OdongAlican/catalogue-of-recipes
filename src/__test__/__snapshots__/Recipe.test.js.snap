// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches recipe snapshot 1`] = `
<div
  className="col-md-3 "
  data-testid="recipe"
>
  <div
    className="card col-md-12 mx-2 mb-2 p-4"
  >
    <div
      className="image-section mx-auto mb-2"
    >
      <img
        alt="boohoo"
        className="image-det"
        src="sample"
      />
    </div>
    <div>
      <h5
        className="text-secondary font-weight-bold"
      >
        title
      </h5>
    </div>
    <p
      className="symbol-paragraph"
    >
       company: 
      <b>
        <i />
      </b>
    </p>
    <label
      className="display-6 visit-us"
      htmlFor="site"
    >
      Visit Us:
      <a
        className="ml-1"
        href="site"
        rel="noopener noreferrer"
        target="_blank"
      >
        site
      </a>
    </label>
    <button
      className="btn btn-primary mb-1"
    >
      <a
        className="text-white"
        href="/recipe/title"
        onClick={[Function]}
      >
        View Details
      </a>
    </button>
  </div>
</div>
`;
